scripts/component-loader.js

```diff
@@ -87,6 +90,9 @@ export default class ComponentLoader {
     let elem = null;
     try {
       elem = await this.createElementAndConfigure(data);
+      elem.webComponentName = this.webComponentName;
```

Comment:
Usually, the good practice is to have the webComponentName as a static part of the component class itself.
So we have it independent of instance or initialisation, here we "set" to avoid having static in all component classes.
If the base component will be under initialisation...

Reply:
But the component base already has the `webComponentName` prop defined.

![2024-08-05 - Capture - 002307@2x](https://github.com/user-attachments/assets/cebd0d23-3bf9-4388-b05e-27797669c758) 

What I was referring here was about all the code that adds the instance to `window.raqnInstances` that it should be inside the component base to cover cases when we create instances by `string` or with `docuemnt.createElement()` or those are not needed to be saved?

```
 elem.webComponentName = this.webComponentName;
 this.instances[elem.componentName] = this.instances[elem.componentName] || [];
 this.instances[elem.componentName].push(elem);
      ```

Reply:
Set Defaults is instance-based as I mentioned.
This should be "static like" property not something set by instance, since is used before instances are created.

Currently, it is set as a getter on the loader.
```
  get webComponentName() {
    return `raqn-${this.componentName.toLowerCase()}`;
  }
  ```
  
  But in theory the each component class should have that info pass to it. or staticly defined at class

Reply:
So usually most frameworks and web components applications has something like

```javascript
export default class Card extends ComponentBase {
  static elementName = 'raqn-card'
  // ...
 }
 ```
 
 Like most frameworks.
 
 
 But the most important part here is that ```window.customElements.define``` is not the scope of the component.
 
and createElement, the usage you shouldn't require access to the component code.

What we need is just a reference at instances and raqnComponents so we reference the original to the handler and list their instances

Reply:
and component Name should be the className it self

Reply:
So for instance
```this.constructor.name``` should be the componentName by it self